Bridges API: fix handling unknown bridge-id

sstrigler opened the pull request:
Fixes [EMQX-9147](https://emqx.atlassian.net/browse/EMQX-9147)

## PR Checklist
Please convert it to a draft if any of the following conditions are not met. Reviewers may skip over until all the items are checked:

- [x] Added tests for the changes
- [x] Changed lines covered in coverage report
- [x] Change log has been added to `changes/{ce,ee}/(feat|perf|fix)-<PR-id>.en.md` and `.zh.md` files
- [x] For internal contributor: there is a jira ticket to track this change
- [-] If there should be document changes, a PR to emqx-docs.git is sent, or a jira ticket is created to follow up
- [-] Schema changes are backward compatible

lafirest commented on changes/ce/fix-10107.zh.md:
```diff
@@ -0,0 +1,6 @@
+对于对 `Bridges API` 的操作，如果 `bridge-id` 未知，我们现在返回 `404`，而不是
```
现在对桥接的 API 进行调用时，如果 `bridge-id` 不存在，将会返回 `404`，而不再是`400`。
然后，还修复了这种情况下，在节点级别上进行 API 调用时，可能导致崩溃的问题。
另外，在启动某个桥接时，会先检查指定桥接是否已启用。
受影响的接口有：
 * [cluster] `/bridges/:id/:operation`,
 * [node] `/nodes/:node/bridges/:id/:operation`,
其中 `operation` 是 `[start|stop|restart]` 之一。

HJianBo commented on changes/ce/fix-10107.zh.md:
```diff
@@ -0,0 +1,6 @@
+对于对 `Bridges API` 的操作，如果 `bridge-id` 未知，我们现在返回 `404`，而不是
+`400`。此外，还修复了一个错误，即如果这是一个节点的操作，会导致崩溃。此外，我们
+现在也检查集群操作 `start`，如果给定的桥被启用。受影响的端点。
+ * [cluster] `/bridges/:id/:operation`,
+ * [node] `/nodes/:node/bridges/:id/:operation`,
+其中 `operation` 是 `[start|stop|restart]` 之一。
```
```suggestion
修复 `Bridges` 相关的 HTTP API 错误：
1. 当请求的 `bridge-id` 未知时，现在返回 `404` 而不是 `400`。
2. 修复对 `nodes/:node/bridges/:id/:operation` 相关的调用，存在的崩溃错误。
3. 当执行 `start` 操作时会检查该桥接是否已经被启用，否则会返回错误。
受影响的路径有：
 * [cluster] `/bridges/:id/:operation`,
 * [node] `/nodes/:node/bridges/:id/:operation`,
其中 `operation` 是 `[start|stop|restart]` 之一。
```

sstrigler replied:
@lafirest this one or yours?

lafirest replied:
this one is also fine, you can use this one.

lafirest commented on changes/ce/fix-10107.en.md:
```diff
@@ -0,0 +1,7 @@
+For operations on `bridges API` if `bridge-id` is unknown we now return `404`
+instead of `400`. Also a bug was fixed that caused a crash if that was a node
+operation. Additionally we now also check for cluster operation `start` if given
+bridge is enabled. Affected endpoints:
```
Personal: Additionally we now also check if the given bridge is enabled when doing the cluster operation `start` .

HJianBo reviewed (dismissed):
LGTM

thalesmg commented on apps/emqx_bridge/src/emqx_bridge_api.erl:
```diff
@@ -524,12 +528,10 @@ schema("/bridges_probe") ->
-                                    RuleIds}
-                            )};
+                        ?BAD_REQUEST(
+                            {<<"Can not delete bridge while active rules defined for this bridge">>,
```
```suggestion
                            {<<"Cannot delete bridge while active rules are defined for this bridge">>,
```

sstrigler commented:
There were issues with code coverage, I changed parts of the code that was untested. 

Also one more minor bug uncovered and fixed. We didn't check for non existent nodes when calling node operations. I've added a line to the changelog files, so please have another look at the translation please.

lafirest commented on changes/ce/fix-10107.en.md:
```diff
@@ -0,0 +1,9 @@
+ * [cluster] `/bridges/:id/:operation`,
+ * [node] `/nodes/:node/bridges/:id/:operation`, where `operation` is one of
+`[start|stop|restart]`.
+Moreover for a node operation, we check if node name is in our cluster and
```
nit: Moreover, for a node operation, we will check if the node is in our cluster, and if not, return `404` instead of `501`.

lafirest commented on changes/ce/fix-10107.zh.md:
```diff
@@ -0,0 +1,8 @@
+ * [cluster] `/bridges/:id/:operation`,
+ * [node] `/nodes/:node/bridges/:id/:operation`,
+其中 `operation` 是 `[start|stop|restart]` 之一。
+此外，对于一个节点操作，我们检查节点名称是否在我们的集群中，并且 返回`404`而不是`501`。
```
此外，对于节点操作，我们将检查节点是否存在于集群中，如果不在，则会返回`404`，而不再是`501`。

JimMoen commented on changes/ce/fix-10107.en.md:
```diff
@@ -0,0 +1,9 @@
+ * [cluster] `/bridges/:id/:operation`,
+ * [node] `/nodes/:node/bridges/:id/:operation`, where `operation` is one of
+`[start|stop|restart]`.
+Moreover, for a node operation, we check if node name is in our cluster and
```
we check ... -> EMQX checks

JimMoen commented on changes/ce/fix-10107.zh.md:
```diff
@@ -0,0 +1,8 @@
+ * [cluster] `/bridges/:id/:operation`,
+ * [node] `/nodes/:node/bridges/:id/:operation`,
+其中 `operation` 是 `[start|stop|restart]` 之一。
+此外，对于节点操作，我们将检查节点是否存在于集群中，如果不在，则会返回`404`，而不再是`501`。
```
我们将检查… -> EMQX 将检查…

thalesmg commented on apps/emqx_bridge/test/emqx_bridge_api_SUITE.erl:
```diff
@@ -612,6 +612,7 @@ t_enable_disable_bridges(Config) ->
         <<"{\"code\":\"BAD_REQUEST\",\"message\":\"Forbidden operation, bridge not enabled\"}">>,
         Res
     ),
+    {ok, 400, Res} = request(post, operation_path(cluster, start, BridgeID), <<"">>),
```
Just a small suggestion: I tried to see in this test (`t_enable_disable_bridges`) if there was an assertion to check that `?BRIDGE_NOT_FOUND/404` is returned for this operation, but I didn't spot it.  Maybe it would be interesting to add a case for that?

sstrigler replied:
Yeah, I might have missed to add one here and there. I'd rather have a cleanup or rather refactoring ticket and make this more composable. 

Current status is that things are very repetitive without clear patterns even. Be it tests or application logic. It's far from ideal. And kinda beyond the scope of this ticket.
